src/main/java/org/elasticsearch/action/admin/indices/upgrade/get/UpgradeStatusResponse.java

```diff
@@ -0,0 +1,191 @@
+        for (String index : indices) {
+            List<ShardUpgradeStatus> shards = Lists.newArrayList();
+            for (ShardUpgradeStatus shard : this.shards) {
+                if (shard.getShardRouting().index().equals(index)) {
+                    shards.add(shard);
+                }
+            }
+            indicesUpgradeStats.put(index, new IndexUpgradeStatus(index, shards.toArray(new ShardUpgradeStatus[shards.size()])));
+        }
+        this.indicesUpgradeStatus = indicesUpgradeStats;
+        return indicesUpgradeStats;
+    }
+
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        shards = new ShardUpgradeStatus[in.readVInt()];
+        for (int i = 0; i < shards.length; i++) {
+            shards[i] = ShardUpgradeStatus.readShardUpgradeStatus(in);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeVInt(shards.length);
+        for (ShardUpgradeStatus shard : shards) {
+            shard.writeTo(out);
+        }
+    }
+
+    public long getTotalBytes() {
+        long totalBytes = 0;
+        for (IndexUpgradeStatus indexShardUpgradeStatus : getIndices().values()) {
+            totalBytes += indexShardUpgradeStatus.getTotalBytes();
+        }
+        return totalBytes;
+    }
+
+    public long getToUpgradeBytes() {
+        long upgradeBytes = 0;
+        for (IndexUpgradeStatus indexShardUpgradeStatus : getIndices().values()) {
+            upgradeBytes += indexShardUpgradeStatus.getToUpgradeBytes();
+        }
+        return upgradeBytes;
+    }
+
+    public long getToUpgradeBytesAncient() {
+        long upgradeBytesAncient = 0;
+        for (IndexUpgradeStatus indexShardUpgradeStatus : getIndices().values()) {
+            upgradeBytesAncient += indexShardUpgradeStatus.getToUpgradeBytesAncient();
+        }
+        return upgradeBytesAncient;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+
+
+        builder.byteSizeField(Fields.SIZE_IN_BYTES, Fields.SIZE, getTotalBytes());
+        builder.byteSizeField(Fields.SIZE_TO_UPGRADE_IN_BYTES, Fields.SIZE_TO_UPGRADE, getToUpgradeBytes());
+        builder.byteSizeField(Fields.SIZE_TO_UPGRADE_ANCIENT_IN_BYTES, Fields.SIZE_TO_UPGRADE_ANCIENT, getToUpgradeBytesAncient());
+
+        String level = params.param("level", "indices");
+        boolean outputShards = "shards".equals(level);
+        boolean outputIndices = "indices".equals(level) || outputShards;
+        if (outputIndices) {
+            builder.startObject(Fields.INDICES);
+            for (IndexUpgradeStatus indexUpgradeStatus : getIndices().values()) {
+                builder.startObject(indexUpgradeStatus.getIndex(), XContentBuilder.FieldCaseConversion.NONE);
+
+                builder.byteSizeField(Fields.SIZE_IN_BYTES, Fields.SIZE, indexUpgradeStatus.getTotalBytes());
```

Comment:
I don't think we should have these at the root level, since they could conflict with index names? Maybe we should have `_all` or something like we do in other apis to represent all the indexes combined?